server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -590,13 +617,34 @@ protected void closeInternal() {
             .addSettingsUpdateConsumer(INDICES_REQUEST_CACHE_MAX_SIZE_ALLOWED_IN_CACHE_SETTING, this::setMaxSizeInRequestCache);
 
         this.defaultMaxMergeAtOnce = CLUSTER_DEFAULT_INDEX_MAX_MERGE_AT_ONCE_SETTING.get(clusterService.getSettings());
+
+        clusterDefaultMaxThreadCount = CLUSTER_DEFAULT_MAX_THREAD_COUNT_SETTING.get(clusterService.getSettings());
+        clusterDefaultMaxMergeCount = CLUSTER_DEFAULT_MAX_MERGE_COUNT_SETTING.get(clusterService.getSettings());
+        clusterDefaultAutoThrottleEnabled = CLUSTER_DEFAULT_AUTO_THROTTLE_SETTING.get(clusterService.getSettings());
+
         clusterService.getClusterSettings()
             .addSettingsUpdateConsumer(CLUSTER_DEFAULT_INDEX_MAX_MERGE_AT_ONCE_SETTING, this::onDefaultMaxMergeAtOnceUpdate);
         clusterService.getClusterSettings()
             .addSettingsUpdateConsumer(
                 MergeSchedulerConfig.CLUSTER_MAX_FORCE_MERGE_MB_PER_SEC_SETTING,
                 this::onClusterLevelForceMergeMBPerSecUpdate
             );
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(
+                this::setClusterDefaultMaxThreadAndMergeCount,
+                List.of(CLUSTER_DEFAULT_MAX_MERGE_COUNT_SETTING, CLUSTER_DEFAULT_MAX_THREAD_COUNT_SETTING),
+                this::validateMaxThreadAndMergeCount
```

Comment:
Lets just have the `this::setClusterDefaultMaxThreadAndMergeCount` called here and make this method 2 steps: 1. validate the value for each index and set it.